In [29]:
import pandas as pd
import numpy as np

# Question 1

In [30]:
data = pd.read_csv('Shopify.csv')

In [31]:
data.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

### a) Think about what could be going wrong with our calculation. Think about a better way to evaluate this data

In [32]:
data['order_amount'].describe()

count      5000.000000
mean       3145.128000
std       41282.539349
min          90.000000
25%         163.000000
50%         284.000000
75%         390.000000
max      704000.000000
Name: order_amount, dtype: float64

#### Mean calculation and Outliers
- Mean of 3145.128 was obtained by taking the average of order amount column.
- Since it is a very large number, it could be influenced by large and small outliers.
- The standard deviation is 41282.539349 is also a large number meaning that the values in the data set are far away from the mean. 
- Shows that the mean is not a robust estimate of AOV.

#### Quartiles and Outliers
- 3rd quartile: 75% of data points fall below 390.00.
- This is substantially lower than the mean 
- Maximum value is 704000.00 which is much higher than the 3rd quarlite.
- It can be conculded that outliers exist and are inflating the AOV.

#### Robust Median
- Median is good to estimate AOV.
- Robust to skeweing by outliers of all sizes.
- Ensures a more accurate assessment of the typical order value.
- Provides a central measure that remains stable despite extreme data points

### (b) What metric would you report for this dataset?

Median

### (c) What is its value

In [33]:
data['order_amount'].median()

284.0

value of median: 284.0

# Question 2

### (a) How many orders were shipped by Speedy Express in total?

In [ ]:
SELECT COUNT(*) FROM Orders
JOIN Shippers ON Shippers.ShipperID = Orders.ShipperID
WHERE Shippers.ShipperName = 'Speedy Express'

54 orders were shipped in total.

### (b) What is the last name of the employee with the most orders?

In [ ]:
SELECT Employees.LastName, Count() AS num_orders FROM Orders
JOIN Employees ON Orders.EmployeeID = Employees.EmployeeID
GROUP BY Employees.LastName ORDER BY num_orders DESC;

The most number of orders was 40 and the last name is Peacock.

### (c) What product was ordered the most by customers in Germany?

In [ ]:
SELECT Products.ProductName, Sum(OrderDetails.Quantity) as most_ordered FROM Orders
JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
JOIN Customers ON Orders.CustomerID = Customers.CustomerID
JOIN Products ON OrderDetails.ProductID = Products.ProductID
WHERE Country = 'Germany'
GROUP BY Products.ProductName ORDER BY most_ordered DESC;

The Boston Crab Meat was ordered 160 times.